pkg/login/social/common.go

```diff
@@ -71,23 +70,24 @@ func HttpGet(client *http.Client, url string) (response HttpGetResponse, err err
 }
 
 func (s *SocialBase) searchJSONForAttr(attributePath string, data []byte) (string, error) {
-	if attributePath == "" {
-		return "", errors.New("no attribute path specified")
-	}
-	if len(data) == 0 {
-		return "", errors.New("empty user info JSON response provided")
+	if attributePath == "" || len(data) == 0 {
```

Comment:
Are both of these errors caused by the user?

Reply:
No, attributePath is by configuration

Reply:
@marefr Thanks. In that case, IMO it's a hard error and we should return an error and log it with error level in the caller (since we for legacy reasons don't bubble).

Reply:
What about the `data` parameter @marefr, is it a hard error if it's empty, or something caused by the client?

Reply:
You can see data is coming from here for example https://github.com/grafana/grafana/pull/22972/files#diff-4225a06af67a76978016c45030173d49R97-R103

Reply:
@marefr So if `data` is empty, will that be a configuration error of the Grafana server? In that case, we should return it as an error. The caller can choose to just error log it.

Reply:
Not really, more strange/bad response from oauth provider

Reply:
@marefr In that case, shouldn't we also return it as an error (that the caller can squash and error log if it wants to)?

Reply:
You're probably right, just don't know what that error should say or what the user should do if getting that error

Reply:
I think for now the caller should just log errors anyway, as we agreed before, so it doesn't really matter.

Reply:
I think both @marefr and I agree that an error should be returned if `attributePath` is empty or if `data` is empty. The caller can squash them and log with error level.